workflow-manager/src/workflow_config.rs

```diff
@@ -411,9 +433,78 @@ impl WorkflowConfig {
 
         Ok(())
     }
+
+    /// Verify our workflow was instantiated successfully
+    async fn verify_init_was_successful(
+        &mut self,
+        connectors: &Connectors,
+        account_instantiate_datas: HashMap<u64, InstantiateAccountData>,
+    ) -> ManagerResult<()> {
+        let mut neutron_connector = connectors.get_or_create_connector(&NEUTRON_DOMAIN).await?;
+        // verify id that is used in workflow is not taken and is not 0
+        ensure!(
+            neutron_connector
+                .query_workflow_registry(NEUTRON_DOMAIN.get_chain_name(), self.id)
+                .await
+                .is_ok(),
+            ManagerError::WorkflowIdAlreadyExists(self.id)
+        );
```

Comment:
We are not checking if it's not 0, right? Change comment/check

Reply:
Fixed